packages/components/src/components/hds/text/types.ts

```diff
@@ -0,0 +1,59 @@
+export type HdsTextTags = keyof HTMLElementTagNameMap;
+
+export enum HdsTextColorValues {
+  Primary = 'primary',
+  Strong = 'strong',
+  Faint = 'faint',
+  Disabled = 'disabled',
+  HighContrast = 'high-contrast',
+  Action = 'action',
+  ActionHover = 'action-hover',
+  ActionActive = 'action-active',
+  Highlight = 'highlight',
+  HighlightOnSurface = 'highlight-on-surface',
+  HighlightHighContrast = 'highlight-high-contrast',
+  Success = 'success',
+  SuccessOnSurface = 'success-on-surface',
+  SuccessHighContrast = 'success-high-contrast',
+  Warning = 'warning',
+  WarningOnSurface = 'warning-on-surface',
+  WarningHighContrast = 'warning-high-contrast',
+  Critical = 'critical',
+  CriticalOnSurface = 'critical-on-surface',
+  CriticalHighContrast = 'critical-high-contrast',
+}
+export type HdsTextColors = `${HdsTextColorValues}`;
+
+export enum HdsTextAlignValues {
+  Left = 'left',
+  Center = 'center',
+  Right = 'right',
+}
+export type HdsTextAligns = `${HdsTextAlignValues}`;
+
+export enum HdsTextWeightValues {
+  Regular = 'regular',
+  Medium = 'medium',
+  Semibold = 'semibold',
+  Bold = 'bold',
+}
+export type HdsTextWeights = `${HdsTextWeightValues}`;
+
+export enum HdsTextSizeValues {
+  FiveHundred = 500,
+  FourHundred = 400,
+  ThreeHundred = 300,
+  TwoHundred = 200,
+  OneHundred = 100,
+}
+type HdsTextSizesString = `${HdsTextSizeValues}`;
+type HdsTextSizesNumber =
+  `${HdsTextSizeValues}` extends `${infer T extends number}` ? T : never;
+export type HdsTextSizes = HdsTextSizesString | HdsTextSizesNumber;
```

Comment:
Checking if the template literal type can be assigned to a template literal type that infers the number, but I'm wondering if we could potentially simplify this as:
```suggestion
type HdsTextSizeNumberValues = HdsTextSizeValues;
type HdsTextSizeStringValues = `${HdsTextSizeValues}`;
export type HdsTextSizes = HdsTextSizeNumberValues | HdsTextSizeStringValues;
```
note that I may be missing some component context here

Reply:
I think that would work out to `Enum | string union`, which would result in consumers only being able to specify the number values via the enum instead of as a number primitive 🤔 

Eg, this would work `@size={{HdsTextSizeNumberValues.OneHundred}}` but this wouldn't `@size={{100))`?

Reply:
you're right, it wouldn't work – carry on!